Calendar/Check-SharingStatus.ps1

```diff
@@ -0,0 +1,326 @@
+    } catch {
+        # Code to run if an error occurs
+        Write-Error "An error occurred: $_"
+    }
+
+    # check if the output is empty
+    if ($null -eq $logOutput.MailboxLog) {
+        Write-Output "No data found for [$Identity]."
+        return
+    }
+
+    $logLines =@();
+    # Split the output into an array of lines
+    $logLines = $logOutput.MailboxLog -split "`r`n"
+
+    # Loop through each line of the output
+    foreach ($line in $logLines) {
+        if ($line -like "*RecipientType*") {
+            $csvString += $line + "`n"
+        }
+    }
+
+    # Clean up output
+    $csvString = $csvString.Replace("Mailbox: ", "")
+    $csvString = $csvString.Replace("Entry MailboxOwner:", "")
+    $csvString = $csvString.Replace("Recipient:", "")
+    $csvString = $csvString.Replace("RecipientType:", "")
+    $csvString = $csvString.Replace("Handler=", "")
+    $csvString = $csvString.Replace("ms-exchange-", "")
+    $csvString = $csvString.Replace("DetailLevel=", "")
+
+    # Convert the CSV string to an object
+    $csvObject = $csvString | ConvertFrom-Csv
+
+    # Access the values as properties of the object
+    foreach ($row in $csvObject) {
+        Write-Debug "$($row.Recipient) - $($row.SharingType) - $($row.detailLevel)"
+    }
+
+    #Filter the output to get the most recent update foreach recipient
+    $mostRecentRecipients = $csvObject | Sort-Object Recipient -Unique | Sort-Object Timestamp -Descending
+
+    # Output the results to the console
+    Write-Output "User [$Identity] has shared their calendar with the following recipients:"
+    Write-Output $mostRecentRecipients | Format-Table -a Timestamp, Recipient, SharingType, DetailLevel
+}
+
+<#
+.SYNOPSIS
+    Formats the AcceptCalendarSharingInvite logs from Export-MailboxDiagnosticLogs for a given identity.
+.DESCRIPTION
+    This function processes calendar sharing invite logs.
+.PARAMETER Identity
+    The SMTP Address for which to process calendar sharing accept logs.
+#>
+function ProcessCalendarSharingAcceptLogs {
+    param (
+        [string]$Identity
+    )
+
+    # Define the header row
+    $header = "Timestamp", "Mailbox", "SharedCalendarOwner", "FolderName"
+    $csvString = @();
+    $csvString = $header -join ","
+    $csvString += "`n"
+
+    Write-Output "------------------------"
+    Write-Output "Looking for Modern Calendar Sharing Accept data for [$Identity]."
+
+    # Call the Export-MailboxDiagnosticLogs cmdlet and store the output in a variable
+    try {
+        # Call the Export-MailboxDiagnosticLogs cmdlet and store the output in a variable
+        # -ErrorAction is not supported on Export-MailboxDiagnosticLogs
+        # $logOutput = Export-MailboxDiagnosticLogs $Identity -ComponentName AcceptCalendarSharingInvite -ErrorAction SilentlyContinue
+
+        $logOutput = Export-MailboxDiagnosticLogs $Identity -ComponentName AcceptCalendarSharingInvite
```

Comment:
Can you email a xml output from this cmdlet and from line 65? We might be able to create something that can be used over and over again a little easier if i am able to see the data.